pkg/api/upgrade_api_test.go

```diff
@@ -0,0 +1,107 @@
+package api_test
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"gotest.tools/assert"
+
+	"helm.sh/helm/v3/pkg/api"
+	"helm.sh/helm/v3/pkg/api/logger"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/release"
+)
+
+type UpgradeTestSuite struct {
+	suite.Suite
+	recorder        *httptest.ResponseRecorder
+	server          *httptest.Server
+	mockUpgrader    *mockUpgrader
+	mockHistory     *mockHistory
+	mockChartLoader *mockChartLoader
+	appConfig       *cli.EnvSettings
+}
+
+func (s *UpgradeTestSuite) SetupSuite() {
+	logger.Setup("default")
+}
+
+func (s *UpgradeTestSuite) SetupTest() {
+	s.recorder = httptest.NewRecorder()
+	s.mockUpgrader = new(mockUpgrader)
+	s.mockHistory = new(mockHistory)
+	s.mockChartLoader = new(mockChartLoader)
+	s.appConfig = &cli.EnvSettings{
+		RepositoryConfig: "./testdata/helm",
+		PluginsDirectory: "./testdata/helm/plugin",
+	}
+	service := api.NewService(s.appConfig, s.mockChartLoader, nil, nil, s.mockUpgrader, s.mockHistory)
+	handler := api.Upgrade(service)
+	s.server = httptest.NewServer(handler)
+}
+
+func (s *UpgradeTestSuite) TestShouldReturnDeployedStatusOnSuccessfulUpgrade() {
+	chartName := "stable/redis-ha"
+	body := fmt.Sprintf(`{
```

Comment:
seems like it's not `gofmt`  ed. use `goimports` as it does gofmt + import checks. golangci would report this as well after enabling test files